modules/fargate/fargate/main.tf

```diff
@@ -0,0 +1,360 @@
+    target_group_arn = aws_lb_target_group.task[each.key].arn
+  }
+
+  condition {
+    field  = lookup(var.containers_definitions[each.key], "rule_field", "host-header")
+    values = lookup(var.containers_definitions[each.key], "rule_values", ["${each.key}.com"])
+  }
+}
+
+resource "aws_lb_listener_rule" "routing_http" {
+  for_each     = var.containers_definitions
+  listener_arn = aws_lb_listener.http.arn
+
+  action {
+    type             = "forward"
+    target_group_arn = aws_lb_target_group.task[each.key].arn
+  }
+
+  condition {
+    field  = lookup(var.containers_definitions[each.key], "rule_field", "host-header")
+    values = lookup(var.containers_definitions[each.key], "rule_values", ["${each.key}.com"])
+  }
+}
+
+resource "aws_lb_target_group" "task" {
+  for_each = var.containers_definitions
+
+  name        = "${each.key}-target-lookup-${lookup(var.containers_definitions[each.key], "task_container_port", "")}"
+  vpc_id      = var.vpc_id
+  protocol    = lookup(var.containers_definitions[each.key], "task_container_protocol", "HTTP")
+  port        = lookup(var.containers_definitions[each.key], "task_container_port", null)
+  target_type = "ip"
+
+  health_check {
+    enabled             = lookup(var.containers_definitions[each.key]["health_check"], "enabled", null)
+    healthy_threshold   = lookup(var.containers_definitions[each.key]["health_check"], "healthy_threshold", null)
+    interval            = lookup(var.containers_definitions[each.key]["health_check"], "interval", null)
+    matcher             = lookup(var.containers_definitions[each.key]["health_check"], "matcher", null)
+    path                = lookup(var.containers_definitions[each.key]["health_check"], "path", null)
+    port                = lookup(var.containers_definitions[each.key]["health_check"], "port", null)
+    protocol            = lookup(var.containers_definitions[each.key]["health_check"], "protocol", null)
+    timeout             = lookup(var.containers_definitions[each.key]["health_check"], "timeout", null)
+    unhealthy_threshold = lookup(var.containers_definitions[each.key]["health_check"], "unhealthy_threshold", null)
+  }
+
+  # NOTE: TF is unable to destroy a target group while a listener is attached,
+  # therefor we have to create a new one before destroying the old. This also means
+  # we have to let it have a random name, and then tag it with the desired name.
+  lifecycle {
+    create_before_destroy = true
+  }
+  tags = merge(
+    var.tags,
+    lookup(var.containers_definitions[each.key], "task_tags", {}),
+    {
+      Name = "${each.key}-target-lookup-${lookup(var.containers_definitions[each.key], "task_container_port", "")}"
+    },
+  )
+}
+
+# ------------------------------------------------------------------------------
+# ECS Task/Service
+# ------------------------------------------------------------------------------
+resource "aws_ecs_task_definition" "task" {
+  for_each = var.containers_definitions
```

Comment:
This is a bit misleading since every item in `container_definitions` will actually get its own task instead of just being one of many containers within a task. I'm not that familiar with ECS Fargate, so perhaps it is a requirement that every task definition can only have a single container definition? If so, should we consider renaming the variable?

Reply:
No, I think you can have multiple containers in one service. The problem is the for_each, if I would be able to iterate multiple times in one resource with different for_each, I think, it could be done. So far I am not sure how to do that.